discord-scripts/thread-management/auto-join.ts

```diff
@@ -57,8 +57,28 @@ async function autoJoinThread(
     }
   }
 
+  // All prefixes of the containing channel name, with dashes converted to
+  // spaces, ordered longest to shortest. For example, #mezo-engineering-musd
+  // would produce ["mezo engineering musd", "mezo engineering", "mezo"].
+  const roleMatchPrefixes = containingChannel?.name
+    .toLowerCase()
+    .split("-")
+    .reduce(
+      (allPrefixes, nameSegment) => [
+        ...allPrefixes,
+        `${allPrefixes.at(-1) ?? []} ${nameSegment}`.trim(),
```

Comment:
Just added `.trim()` here since on local test, roles were being outputted in some cases `role here, role here is, role here there,` otherwise let's merge!